gnosis-buyback/contracts/Buyback.sol

```diff
@@ -0,0 +1,539 @@
+            _auctionIndexes, 
+            _auctionAmounts
+        );
+    }
+
+    /**
+     * @notice postSellOrder approve trading
+     * @param _buybackId buyback id
+     */
+    function postSellOrder(uint _buybackId) public {
+        Buyback storage userBuyback = buybacks[_buybackId];
+
+        uint[] memory auctionIndexes = userBuyback.auctionIndexes;
+        address userAddress = userBuyback.userAddress;
+        address sellToken = userBuyback.sellToken;
+        uint tipAmount = userBuyback.tipAmount;
+
+        require(auctionIndexes.length > 0, "buyback doesn't exist");
+        // ensure buyback has not  been executed
+        require(userBuyback.hasExecutedBuyback == false, "buyback has been executed");
+        // check if buyback has expired
+        require(isTimePassed(userBuyback.expires), "buyback has expired");
+        require(hasEnoughBalance(_buybackId, auctionIndexes, userAddress, sellToken), "user does not have enough balance to post sell order");
+        require(hasEnoughTippingBalance(userAddress, tipAmount), "user does not have enough tipping balance");
+
+
+        uint newBuyerBalance;
+        uint dxIndex;
+        uint totalAmount = 0;
+
+        address buyToken = userBuyback.buyToken;        
+        uint userSellTokenBalance = balances[userAddress][sellToken];
+
+        for( uint i = 0; i < auctionIndexes.length; i++ ) {
+            // get the auction amount for that index
+            uint amount = auctionIndexWithAmount[_buybackId][auctionIndexes[i]]; 
+            totalAmount = totalAmount.add(amount);
+
+            // check if the user as enough balance
+            require(userSellTokenBalance >= amount, "user does not have enough balance"); 
+
+            approveDutchX(sellToken, amount);
+            depositDutchx(sellToken, amount);
+
+            (dxIndex, newBuyerBalance) = dx.postSellOrder(sellToken, buyToken, auctionIndexes[i], amount);
+
+            // maps the auction index to the auction index from dx auction
+            dxAuctionIndex[_buybackId][auctionIndexes[i]] = dxIndex;
+            userSellTokenBalance = userSellTokenBalance.sub(amount);
+
+            balances[userAddress][sellToken] = userSellTokenBalance;
+        }
+
+        // tip the user that poked the postSellOrder function
+        tip(userAddress, tipAmount, msg.sender);
+        // set user executed buyback true
+        userBuyback.hasExecutedBuyback = true;
+
+        userBuybackAmountTotal[userAddress][sellToken] = userBuybackAmountTotal[userAddress][sellToken].sub(totalAmount);
+        userTipAmountTotal[userAddress] = userTipAmountTotal[userAddress].sub(tipAmount);
+
+        emit PostSellOrder(
+            _buybackId,
+            userAddress,
+            buyToken, 
+            sellToken, 
+            userSellTokenBalance
+        );
+ 
+    }
+
+    /**
+     * @notice updateDutchExchange
+     */
+    function updateDutchExchange(DutchExchange _dx) external onlyOwner {
+        dx = _dx;
```

Comment:
Security wise, this one is very sensitive. 

The owner could replace the DutchX with a contract that implements the deposit and the sell, but steals the tokens.

Do you see any solution to mitigate this?

Reply:
I have tried checking but the assumption would be that solidity should prevent updating it with anything that doesn't have the DuctchExchange abi

Reply:
i have removed the function